shotover-proxy/tests/cassandra_int_tests/mod.rs

```diff
@@ -914,3 +914,102 @@ async fn test_protocol_v5_compression_encode(#[case] driver: CassandraDriver) {
         shotover.shutdown_and_then_consume_events(&[]).await;
     }
 }
+
+#[rstest]
+#[case::cdrs(CdrsTokio)]
+#[tokio::test(flavor = "multi_thread")]
+#[serial]
+async fn test_protocol_v5_non_self_contained(#[case] driver: CassandraDriver) {
+    use rand::Rng;
+
+    let _docker_compose =
+        docker_compose("tests/test-configs/cassandra-passthrough/docker-compose.yaml");
+
+    let shotover = ShotoverProcessBuilder::new_with_topology(
+        "tests/test-configs/cassandra-passthrough/topology-encode.yaml",
+    )
+    .start()
+    .await;
+
+    let connection_creator = || {
+        CassandraConnectionBuilder::new("127.0.0.1", 9042, driver)
+            .with_protocol_version(ProtocolVersion::V5)
+            .build()
+    };
+
+    let connection = connection_creator().await;
+
+    connection.execute(
+        "CREATE KEYSPACE keyspace_name WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };",
+    ).await;
+
+    connection
+        .execute("CREATE TABLE keyspace_name.my_table (id INT PRIMARY KEY, data BLOB);")
+        .await;
+
+    let random_bytes = rand::thread_rng().gen::<[u8; 32]>();
+
+    let mut queries = Vec::new();
+    for id in 0..1000 {
+        let statement = format!(
+            "INSERT INTO keyspace_name.my_table (id, data) VALUES ({}, 0x{})",
+            id,
+            hex::encode(random_bytes)
+        );
+        queries.push(statement);
+    }
+
+    connection.execute_batch(queries).await;
```

Comment:
how did you verify this actually triggers the PAYLOAD_SIZE_LIMIT?
Also can this just be a test case in `standard_test_suite` so that we always test and dont have to pay for more database startup/shutdowns?

Reply:
Adding the test to the current main branch shows that the test fails because it is above the payload max size, and the driver has sent it as a non self contained frame.


![image](https://user-images.githubusercontent.com/39339036/235392150-6ac6a4b6-90fc-45c9-9d62-ff0611ff6704.png)

![image](https://user-images.githubusercontent.com/39339036/235392179-cca6cba3-d3ea-44c3-ad09-c25cd6624157.png)